### DISCUSSION TITLE
Use aloam in singularity

### DISCUSSION THREAD

**Discussion by lidiaxp** [ctu-mrs/mrs_uav_system#172] [Q&A] [answered]
Hi, 

I am trying to use Aloam in singularity, but the host machine is an Ubuntu 22.04. The instructions to use Aloam is to run the install.sh, but I can't do it outside the container because it is not compatible with Ubuntu 22.04, and can't do it inside the container due several errors about permissions. I also, tried to run the container with root permission (KEEP_ROOT_PRIVS=true) but the /tmp folder to install the cerves solver (a dependency to aloam) is just to read.

Can I just change how the /tmp folder to readable and writable, or is it going to causes more problems (considering I am using Ubuntu 22.04)? Otherwise, is there another way to intall aloam with singularity?

Ps. I also tried to install by `sudo apt install ros-noetic-mrs-aloam-core`, but it appears the same errors about permission.

Thank you very much for any help.

**Comment by klaxalk** [comment, marked as answer]
Hey Lidia, have you tried to just modify the [recipe](https://github.com/ctu-mrs/mrs_singularity/blob/master/recipes/stable/recipe.def) by adding the `apt-get -y install ros-noetic-mrs-aloam-core` there? That should be the simplest way. Alternatively, you can modify an existing image: https://github.com/ctu-mrs/mrs_singularity/#modifying-an-existing-container-possible-but-not-recommended

Tomas

**Reply by lidiaxp** [reply]
Hi Tomas, I added `apt-get -y install ros-noetic-mrs-aloam-core` to recipes now and it worked. Thank you very much.
